### PR TITLE
set state properly for timing event modal

### PR DESCRIPTION
**Pull request by denisdimitrov**
The issue was not the form validation but the area that selects and sets the state. The problem was the callback being passed to the children div which was only a text, and clicking outside the text wasn't changing the state and the validation was failing.

This might happen in other forms as well and IMO we should pass an array of options to the dropdown instead of children, that's the way React Native handles dropdowns and lists and it works without any problems at all.

Fixes: https://github.com/sparkletown/internal-sparkle-issues/issues/1811

### CODE REVIEW

File: src/components/organisms/TimingEventModal/TimingEventModal.tsx [hunk 1]
```diff
@@ -165,7 +172,9 @@ export const TimingEventModal: React.FC<TimingEventModalProps> = ({
           {!eventSpace?.name && (
             <>
               <div className="mb-6">
-                <Dropdown title={"None"}>{renderedSpaceIds}</Dropdown>
+                <Dropdown onSelect={selectSpace} title={"None"}>
```
Comment (goran-peoski-work) [inline src/components/organisms/TimingEventModal/TimingEventModal.tsx]:
```suggestion
                <Dropdown onSelect={selectSpace} title="None">
```

File: src/components/organisms/TimingEventModal/TimingEventModal.tsx [hunk 2]
```diff
@@ -103,14 +104,20 @@ export const TimingEventModal: React.FC<TimingEventModalProps> = ({
...
         );
       }) ?? [],
     [spacesMap]
   );
 
+  const selectSpace = (option: Option) => {
+    if (option.value) {
+      setSelectedSpace(option.value as SpaceType);
+    }
+  };
```
Comment (goran-peoski-work) [inline src/components/organisms/TimingEventModal/TimingEventModal.tsx]: [Q] Does this mean once selected a space can't be un-selected? How does or should one be able to remove any space from the options?

A second, maybe related, scenario: user picks a space, later deletes that space, what id shown here?

**Review by goran-peoski-work — COMMENTED** [review]
It seems to be resolving an issue, but unsure if causing new ones

File: src/components/organisms/TimingEventModal/TimingEventModal.tsx (hunk 2 above)
Comment (denisdimitrov) [reply src/components/organisms/TimingEventModal/TimingEventModal.tsx]: The space is required and once selected it makes no sense to unselect it.

We haven't handled this delete case or any other similar ones on the admin side.

Comment (goran-peoski-work) [reply src/components/organisms/TimingEventModal/TimingEventModal.tsx]: good enough

**Review by goran-peoski-work — APPROVED** [review]
LGTM

**Pull request merged**